exporters/otlp/common/src/main/java/io/opentelemetry/exporter/otlp/internal/MetricAdapter.java

```diff
@@ -269,10 +285,48 @@ private static AggregationTemporality mapToTemporality(
           .getAttributes()
           .forEach(
               (key, value) -> builder.addAttributes(CommonAdapter.toProtoAttribute(key, value)));
+      doubleHistogramPoint.getExemplars().forEach(e -> builder.addExemplars(toExemplar(e)));
       result.add(builder.build());
     }
     return result;
   }
 
+  static Exemplar toExemplar(io.opentelemetry.sdk.metrics.data.Exemplar exemplar) {
+    // TODO - Use a thread local cache for spanid/traceid -> byte conversion.
+    Exemplar.Builder builder = Exemplar.newBuilder();
+    builder.setTimeUnixNano(exemplar.getRecordTimeNanos());
+    if (exemplar.getSpanId() != null) {
+      builder.setSpanId(convertSpanId(exemplar.getSpanId()));
+    }
+    if (exemplar.getTraceId() != null) {
+      builder.setTraceId(convertTraceId(exemplar.getTraceId()));
+    }
+    exemplar
+        .getFilteredAttributes()
+        .forEach(
+            (key, value) ->
+                builder.addFilteredAttributes(CommonAdapter.toProtoAttribute(key, value)));
+    if (exemplar instanceof LongExemplar) {
+      builder.setAsInt(((LongExemplar) exemplar).getValue());
```

Comment:
Two things:
1. I've been told this project is trying to avoid `.stream()` overhead, maybe @anuraaga or @jkwatson can comment?
2. This is a protocol buffer builder, I can't touch how it works (it's auto-generated code) :)

Reply:
Got it, could you please mark these comments as resolved (this and the previous two)?